app/models/user.rb

```diff
@@ -0,0 +1,23 @@
+class User < ApplicationRecord
+  has_many :orders, dependent: :destroy
+  has_one :cart, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+
+  enum role: { admin: 0, user: 1 }
+
+  has_secure_password
+
+  validates :name, presence: true
+  validates :email, presence: true, uniqueness: true, format: { with: URI::MailTo::EMAIL_REGEXP }
```

Comment:
Is `user@admin` a valid email in this regex?

Reply:
Yes bhai. Should I change it?